CMakeLists.txt

```diff
@@ -214,12 +214,6 @@ set(OTELCPP_PROTO_PATH
     CACHE PATH "Path to opentelemetry-proto")
 
 if(WIN32)
-  if(BUILD_TESTING)
-    if(MSVC)
-      # Warning as error: warning STL4036: <ciso646> is removed in C++20
-      add_compile_options(/wd4996)
```

Comment:
Why is this option no longer needed?

Reply:
The opentelemetry-cpp code does not even include header ciso464, and the build passes in maintainer mode on windows without the option, even when tests are compiled, with the dependency in GTest.

PR#2114 added this, but I think it was not needed.

This fix corrects it, and resolves you comment:

https://github.com/open-telemetry/opentelemetry-cpp/pull/2114/files#r1260156709